checks.d/disk.py

```diff
@@ -0,0 +1,229 @@
+# stdlib
+import os
+import re
+
+# 3p
+try:
+    import psutil
+except ImportError:
+    psutil = None
+
+# project
+from checks import AgentCheck
+from config import _is_affirmative
+from util import Platform
+import utils.subprocess_output
+
+
+class Disk(AgentCheck):
+    """ Collects metrics about the machine's disks. """
+    # -T for filesystem info
+    DF_COMMAND = ['df', '-T']
+    FAKE_DEVICES = ['udev', 'sysfs', 'rpc_pipefs', 'proc', 'devpts']
+    METRIC_DISK = 'system.disk.{0}'
+    METRIC_INODE = 'system.fs.inodes.{0}'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        if instances is not None and len(instances) > 1:
+            raise Exception("Disk check only supports one configured instance.")
+        AgentCheck.__init__(self, name, init_config,
+                            agentConfig, instances=instances)
+
+    def check(self, instance):
+        """Get disk space/inode stats"""
+        # First get the configuration.
+        self._load_conf(instance)
+
+        # Windows and Mac will always have psutil
+        # (we have packaged for both of them)
+        if self._psutil():
+            self.collect_metrics_psutil()
+        else:
+            self.collect_metrics_manually()
+
+    def _psutil(self):
+        return psutil is not None
+
+    def _load_conf(self, instance):
+        self._excluded_filesystems = instance.get('excluded_filesystems', [])
+        self._excluded_disks = instance.get('excluded_disks', [])
+        self._tag_by_filesystem = _is_affirmative(
+            instance.get('tag_by_filesystem', ''))
+        self._all_partitions = _is_affirmative(
+            instance.get('all_partitions', 'yes'))
```

Comment:
Just for clarity here I'd recommend using a boolean directly rather than a string for the default.